plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt

```diff
@@ -112,6 +112,11 @@ class FeatureDevSessionContext(val project: Project, val maxProjectSizeBytes: Lo
     suspend fun ignoreFile(file: VirtualFile): Boolean = ignoreFile(file.path)
 
     suspend fun ignoreFile(path: String): Boolean {
+        // explicitly allow the Gradle wrapper JAR file
+        if (path.endsWith("gradle/wrapper/gradle-wrapper.jar")) {
```

Comment:
don't hardcode here. add a list of them at the top. you can say something like `private val requiredBinaryFilesForExecution = setOf("gradle/wrapper/gradle-wrapper.jar")`